### ISSUE TITLE
generate_chrom_lens.py can't find xml.sax.saxutils when called from fusioncatcher-build

### ISSUE THREAD

**Issue by GodloveD** [ndaniel/fusioncatcher#39]
I just downloaded and installed 0.99.7b like so:  
```
wget http://sf.net/projects/fusioncatcher/files/bootstrap.py -O bootstrap.py && python bootstrap.py -t --download -y
```
Dependencies are found, all messages are happy and everyone feels good.  Then I try to run `fusioncatcher-build` and I receive an error like so:
```
$ mkdir tmp
$ fusioncatcher-build -g canis_familiaris -o ./tmp
WARNING: Cannot restart automatically because the previous log file '/scratch/godlovedc/tmp/fusioncatcher-build.log' cannot be found!
The workflow will be restarted from the beginning with step 1!
Log of the pipeline:
--------------------------------------------------------------------------------
Starting execution with step 1.
////////////////////////////////////////////////////////////////////////////////
  Running: step = 1   Time: 16:21   Date: 2017-03-21 (elapsed time: 0d:0h:0m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/scratch/godlovedc'
python_version.py
--------------------------------------------------------------------------------
+-->EXECUTING...
Python version: 2.7.13 |Continuum Analytics, Inc.| (default, Dec 20 2016, 23:09:15)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
Python executable: /usr/local/bin/python
--------------------------------------------------------------------------------
==> Execution time: 0 day(s), 0 hour(s), 0 minute(s), and 0 second(s)
////////////////////////////////////////////////////////////////////////////////
  Running: step = 2   Time: 16:21   Date: 2017-03-21 (elapsed time: 0d:0h:0m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/scratch/godlovedc'
biopython_version.py
--------------------------------------------------------------------------------
+-->EXECUTING...
1.68
--------------------------------------------------------------------------------
==> Execution time: 0 day(s), 0 hour(s), 0 minute(s), and 0 second(s)
////////////////////////////////////////////////////////////////////////////////
  Running: step = 3   Time: 16:21   Date: 2017-03-21 (elapsed time: 0d:0h:0m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/scratch/godlovedc'
printf \
"canis_familiaris"  \
> /scratch/godlovedc/tmp/organism.txt
--------------------------------------------------------------------------------
+-->EXECUTING...
--------------------------------------------------------------------------------
==> Execution time: 0 day(s), 0 hour(s), 0 minute(s), and 0 second(s)
////////////////////////////////////////////////////////////////////////////////
  Running: step = 4   Time: 16:21   Date: 2017-03-21 (elapsed time: 0d:0h:0m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/scratch/godlovedc'
get_genome.py \
--organism canis_familiaris \
--server ftp.ensembl.org \
--output /scratch/godlovedc/tmp/
--------------------------------------------------------------------------------
+-->EXECUTING...
Downloading the genome of organism 'canis_familiaris' from Ensembl!
230 Anonymous access granted, restrictions apply
Downloading: /pub/current_fasta//canis_familiaris/dna/Canis_familiaris.CanFam3.1.dna.chromosome.MT.fa.gz
Downloading: /pub/current_fasta//canis_familiaris/dna/Canis_familiaris.CanFam3.1.dna.toplevel.fa.gz
Executing:   gzip -d -f -c '/scratch/godlovedc/tmp/Canis_familiaris.CanFam3.1.dna.chromosome.MT.fa.gz' > '/scratch/godlovedc/tmp/Canis_familiaris.CanFam3.1.dna.chromosome.MT.fa'
Executing:   gzip -d -f -c '/scratch/godlovedc/tmp/Canis_familiaris.CanFam3.1.dna.toplevel.fa.gz' > '/scratch/godlovedc/tmp/Canis_familiaris.CanFam3.1.dna.toplevel.fa'
--------------------------------------------------------------------------------
==> Execution time: 0 day(s), 0 hour(s), 0 minute(s), and 56 second(s)
////////////////////////////////////////////////////////////////////////////////
  Running: step = 5   Time: 16:22   Date: 2017-03-21 (elapsed time: 0d:0h:1m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/scratch/godlovedc'
generate_chrom_lens.py \
--input_genome /scratch/godlovedc/tmp/genome.fa \
--output /scratch/godlovedc/tmp/
--------------------------------------------------------------------------------
+-->EXECUTING...
Traceback (most recent call last):
  File "/usr/local/apps/fusioncatcher/0.99.7b/fusioncatcher/bin/generate_chrom_lens.py", line 67, in <module>
    import Bio.SeqIO
  File "/usr/local/python/lib/python2.7/site-packages/Bio/SeqIO/__init__.py", line 374, in <module>
    from . import SeqXmlIO
  File "/usr/local/python/lib/python2.7/site-packages/Bio/SeqIO/SeqXmlIO.py", line 17, in <module>
    from xml.sax.saxutils import XMLGenerator
ImportError: No module named sax.saxutils


ERROR: Workflow execution failed at step 5 while executing:
----------------
   generate_chrom_lens.py \
   --input_genome /scratch/godlovedc/tmp/genome.fa \
   --output /scratch/godlovedc/tmp/
----------------


Executing second time the same step/command in order to capture error messages (i.e. STDERR)...

-------------------------------------------
Traceback (most recent call last):
  File "/usr/local/apps/fusioncatcher/0.99.7b/fusioncatcher/bin/generate_chrom_lens.py", line 67, in <module>
    import Bio.SeqIO
  File "/usr/local/python/lib/python2.7/site-packages/Bio/SeqIO/__init__.py", line 374, in <module>
    from . import SeqXmlIO
  File "/usr/local/python/lib/python2.7/site-packages/Bio/SeqIO/SeqXmlIO.py", line 17, in <module>
    from xml.sax.saxutils import XMLGenerator
ImportError: No module named sax.saxutils


################################################################################
################################################################################
TOTAL RUNNING TIME: 0 day(s), 0 hour(s), 0 minute(s), and 57 second(s)
################################################################################
################################################################################
```
So it seems like an error importing `BioSeqIO` because it can't find `sax.saxutils`.  But, if I try importing that module myself, I don't have any problems:
```
$ python
Python 2.7.13 |Continuum Analytics, Inc.| (default, Dec 20 2016, 23:09:15)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import Bio.SeqIO
>>> exit()
```
And even stranger, if I just run `generate_chrom_lens.py` without the `fusioncatcher-build` wrapper, I don't have any problems.
```
$    generate_chrom_lens.py \
>    --input_genome /scratch/godlovedc/tmp/genome.fa \
>    --output /scratch/godlovedc/tmp/
  * chromosome 1 has length 122678785
  * chromosome 10 has length 69331447
  * chromosome 11 has length 74389097
  * chromosome 12 has length 72498081
  * chromosome 13 has length 63241923
  * chromosome 14 has length 60966679
  * chromosome 15 has length 64190966
  * chromosome 16 has length 59632846
  * chromosome 17 has length 64289059
  * chromosome 18 has length 55844845
  * chromosome 19 has length 53741614
  * chromosome 2 has length 85426708
  * chromosome 20 has length 58134056
  * chromosome 21 has length 50858623
  * chromosome 22 has length 61439934
  * chromosome 23 has length 52294480
  * chromosome 24 has length 47698779
  * chromosome 25 has length 51628933
  * chromosome 26 has length 38964690
  * chromosome 27 has length 45876710
  * chromosome 28 has length 41182112
  * chromosome 29 has length 41845238
  * chromosome 3 has length 91889043
  * chromosome 30 has length 40214260
  * chromosome 31 has length 39895921
  * chromosome 32 has length 38810281
  * chromosome 33 has length 31377067
  * chromosome 34 has length 42124431
  * chromosome 35 has length 26524999
  * chromosome 36 has length 30810995
  * chromosome 37 has length 30902991
  * chromosome 38 has length 23914537
  * chromosome 4 has length 88276631
  * chromosome 5 has length 88915250
  * chromosome 6 has length 77573801
  * chromosome 7 has length 80974532
  * chromosome 8 has length 74330416
  * chromosome 9 has length 61074082
  * chromosome MT has length 16727
  * chromosome X has length 123869142
  * chromosome JH373233.1 has length 2660953
  * chromosome JH373234.1 has length 1881673
  * chromosome JH373235.1 has length 1415205
  * chromosome JH373236.1 has length 1067467
[...snip...]
  * chromosome AAEX03026066.1 has length 2046
  * chromosome AAEX03026067.1 has length 1999
  * chromosome AAEX03026068.1 has length 1819
  * chromosome AAEX03026069.1 has length 1790
  * chromosome AAEX03026070.1 has length 1685
  * chromosome AAEX03026071.1 has length 1573
  * chromosome AAEX03026072.1 has length 1115
$
```
What's going on?

**Comment by ndaniel** [comment]
Hi!

According to these: 

- https://groups.google.com/forum/#!topic/django-users/8uJHp-hScXA 
- https://github.com/sphinx-doc/sphinx/issues/3201
- [Google search "ImportError: No module named sax.saxutils"](https://www.google.fi/search?espv=2&q="ImportError%3A+No+module+named+sax.saxutils"&oq="ImportError%3A+No+module+named+sax.saxutils)

it looks like some conflict between different Python libraries is going on.

A quickaround this, now that you already have run manully this
```
$    generate_chrom_lens.py \
>    --input_genome /scratch/godlovedc/tmp/genome.fa \
>    --output /scratch/godlovedc/tmp/
```
would be to re-start fusioncatcher-build.py from step 6, that is something like this
```
fusioncatcher-build -g canis_familiaris -o ./tmp --start 6
```

**Comment by GodloveD** [comment]
I was able to alternate between running `fusioncatcher-build ... --start failed_step+1` and manually running the failed command for a while and I advanced to step 111.  Now I receive a new error.
```
////////////////////////////////////////////////////////////////////////////////
  Running: step = 111   Time: 10:10   Date: 2017-03-22 (elapsed time: 0d:0h:0m)
\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
==> Current working directory: '/scratch/godlovedc'
bowtie-build \
-f  \
--quiet  \
--offrate 1 \
--ftabchars 7 \
 /scratch/godlovedc/tmp/rtrna_mt.fa \
 /scratch/godlovedc/tmp/rtrna_mt_index/
--------------------------------------------------------------------------------
+-->EXECUTING...
bowtie-build: error while loading shared libraries: libtbbmalloc_proxy.so.2: cannot open shared object file: No such file or directory


ERROR: Workflow execution failed at step 111 while executing:
----------------
   bowtie-build \
   -f  \
   --quiet  \
   --offrate 1 \
   --ftabchars 7 \
    /scratch/godlovedc/tmp/rtrna_mt.fa \
    /scratch/godlovedc/tmp/rtrna_mt_index/
----------------
Please, check if 'Bowtie' (from <http://bowtie-bio.sourceforge.net/index.shtml>) is installed correctly and it is in the corresponding PATH (or if 'configuration.cfg' file is set up correctly)!
  * Size '/scratch/godlovedc/tmp/rtrna_mt.fa' = 125332 bytes
  * Size '/scratch/godlovedc/tmp/rtrna_mt_index/' = 0 bytes


Executing second time the same step/command in order to capture error messages (i.e. STDERR)...

-------------------------------------------
bowtie-build: error while loading shared libraries: libtbbmalloc_proxy.so.2: cannot open shared object file: No such file or directory
```

**Comment by GodloveD** [comment]
Editing the `fusioncatcher/etc/configuration.cfg` as you suggest does fix my earlier issues with the python path, but my bowtie errors persist.

**Comment by ndaniel** [comment]
The bowtie error from above looks like is related to this https://github.com/ndaniel/fusioncatcher/issues/20 which was fixed by running this on Ubuntu:
```
sudo apt-get install libtbb-dev libtbb2
```
I guess that on RedHat replacing apt-get with yum might do the trick.
Does this fix it?

**Comment by GodloveD** [comment]
Thanks for the suggestion.  It actually looks like the bowtie version that is packaged with fusioncatcher was compiled against a newer glibc.  (We are still running RHEL6/CentOS6 and our glibc is a bit older.)  The solution was to re-compile bowtie on our system against our old glibc, and copy the executables to the tools subdirectory.  Between editing `configuration.cfg` and recompiling bowtie, I now have an installation capable of running the entire `fusioncatcher-build` pipeline (143 steps) without error.  Thanks.

**Comment by ndaniel** [comment]
Ok.

Just one comment, the FusionCatcher's installation script downloads and installs BOWTIE from the official BOWTIE's site, that is: 
https://sourceforge.net/projects/bowtie-bio/files/bowtie/1.2.0/bowtie-1.2-linux-x86_64.zip

**Comment by ndaniel** [comment]
This has been fixed in v0.99.7c. If you have v0.99.7c and still have this issue then please re-install FusionCatcher again and this will solve it!
